examples/proposed/counting-semaphore.w

```diff
@@ -0,0 +1,75 @@
+bring cloud;
+
+struct CountingSemaphoreProps {
+  availableResources: num;
+}
+
+resource CountingSemaphore {
+  public limit: num;
+  _counter: cloud.Counter;
+
+  // need some ttl solution here,
+  // probably in-house once replaced with a key-value store
+  init(props: CountingSemaphoreProps) {
+    // pseudocode: input validation
+    this.limit = props.availableResources;
+    this._counter = new cloud.Counter();
+  }
+
```

Comment:
for some reason I wanted to do something below from the spec, but none of them works with the current complier.

let me know if they are expected or not - unlike the `struct` one, it has a nice message saying WIP.

```
public inflight var my_variable: bool; // <- `inflight` and `var` do not seem to play together as of now
inflight init() {} // <- got complaints about have `init()` defined already, which, in the context, is a preflight - like the kitchen sink example
```

Reply:
I think these are both compiler bugs

Reply:
bug reported: https://github.com/winglang/wing/issues/1271